### DISCUSSION TITLE
OpenSSL_1_1_1-stable vs. openssl-3.0

### DISCUSSION THREAD

**Discussion by juha-h** [baresip/baresip#2290] [Q&A]
According to https://www.openssl.org/source/, there is two LTS OpenSSL versions: OpenSSL_1_1_1-stable and openssl-3.0. I have been using the former.  Would also openssl-3.0 work OK with baresip?

**Comment by sreimers** [comment]
Yes openssl-3.0 works fine, using in production already.

**Comment by alfredh** [comment]
You should wait for openssl 3.0.7 which is due 1. Nov and contains fixed for known issues.

OpenSSL version 3.0.x all contain the undisclosed issue.

**Comment by juha-h** [comment]
Alfred E. Heggestad writes:

> You should wait for openssl 3.0.7 which is due 1. Nov and contains
> fixed for known issues. 
> 
> OpenSSL version 3.0.x all contain the undisclosed issue.

Yes, I read about it and then started to wonder if I should after the
fix switch to openssl-3.0.  Good to know that openssl-3.0 is OK.

**Comment by juha-h** [comment]
I build openssl-3.0 libraries and then tried to build baresip (in Android Studio).  ld failed with error:
```
 ld: error: undefined symbol: SSL_get_peer_certificate
```
Then I found this in the sources:
```
./doc/man3/SSL_get_peer_certificate.pod:SSL_get_peer_certificate() was deprecated in 3.0.0.
```
So looks like it needs to be replaced by something else.

**Reply by sreimers** [reply]
Looks like you build openssl with `no-deprecated`, try to remove this option. We have a PR https://github.com/baresip/re/pull/585 for next release.

**Comment by juha-h** [comment]
Thanks, build now succeeded.
